lib/iris/loading.py

```diff
@@ -15,6 +16,39 @@
 from iris.warnings import IrisLoadWarning
 
 
+class _ConcreteDerivedLoading(threading.local):
+    """A thread-safe state for controlling the laziness of loaded coordinates.
```

Comment:
```suggestion
    """An object to control whether factory references are loaded with lazy or real data.
```
what I'm trying to address here:
  * it applies specifically to factory references, not to coordinates in general (which *are* lazy)
  * I don't think we need to highlight the thread-safe nature.